src/ray/core_worker/core_worker.cc

```diff
@@ -1938,33 +1941,19 @@ CoreWorker::ListNamedActors(bool all_namespaces) {
 
   // This call needs to be blocking because we can't return until we get the
   // response from the RPC.
-  std::shared_ptr<std::promise<void>> ready_promise =
-      std::make_shared<std::promise<void>>(std::promise<void>());
   const auto &ray_namespace = job_config_->ray_namespace();
-  RAY_CHECK_OK(gcs_client_->Actors().AsyncListNamedActors(
-      all_namespaces, ray_namespace,
-      [&actors, ready_promise](const std::vector<rpc::NamedActorInfo> &result) {
-        for (const auto &actor_info : result) {
-          actors.push_back(std::make_pair(actor_info.ray_namespace(), actor_info.name()));
-        }
-        ready_promise->set_value();
-      }));
-
-  // Block until the RPC completes. Set a timeout to avoid hangs if the
-  // GCS service crashes.
-  Status status;
-  if (ready_promise->get_future().wait_for(std::chrono::seconds(
-          RayConfig::instance().gcs_server_request_timeout_seconds())) !=
-      std::future_status::ready) {
+  const auto status =
+      gcs_client_->Actors().SyncListNamedActors(all_namespaces, ray_namespace, actors);
+  if (status.IsTimedOut()) {
     std::ostringstream stream;
     stream << "There was timeout in getting the list of named actors, "
               "probably because the GCS server is dead or under high load .";
     return std::make_pair(actors, Status::TimedOut(stream.str()));
+  } else if (!status.ok()) {
```

Comment:
```
 } else {
    return std::make_pair(actors, status);
}
```
btw, for the return with actors, you can always move them, for example:

`return make_pair(std::move(actors), status));`